python/sglang/srt/models/deepseek_v2.py

```diff
@@ -646,10 +745,18 @@ def __init__(
         self.config = config
         self.quant_config = quant_config
         self.model = DeepseekV2Model(config, cache_config, quant_config)
-        self.lm_head = ParallelLMHead(
-            config.vocab_size, config.hidden_size, quant_config=quant_config
-        )
-        self.logits_processor = LogitsProcessor(config)
+        if global_server_args_dict["enable_dp_mla"]:
+            self.lm_head = ReplicatedLinear(
```

Comment:
why cannot we parallelize this?

Reply:
Because the data in each worker is different. We need to all-gather data before the TP and slice data after it. Extra operations and communication overhead are added.